crates/reporting/src/error/type.rs

```diff
@@ -1645,6 +1645,44 @@ fn to_expr_report<'b>(
                 unimplemented!("record default field is not implemented yet")
             }
             Reason::ImportParams(_) => unreachable!(),
+            Reason::FunctionOutput => {
+                let problem = alloc.concat([
+                    alloc.text("This "),
+                    alloc.keyword("return"),
+                    alloc.reflow(
+                        " statement doesn't match the return type of its enclosing function:",
+                    ),
+                ]);
+
+                let comparison = type_comparison(
+                    alloc,
+                    found,
+                    expected_type,
+                    ExpectationContext::Arbitrary,
+                    add_category(alloc, alloc.text("It is"), &category),
+                    alloc.concat([
+                        alloc.reflow("But I need every "),
+                        alloc.keyword("return"),
+                        alloc.reflow(" statement in that function to return:"),
```

Comment:
minor: this should probably say "but I need every branch of this function..."

Reply:
I see what you're getting at, but I think this is misleading when we have `if` expressions/statements and early returns. For example, in this function:

```roc
myFunc = \x ->
    y =
        if x == 5 then
            return "early return"
        else
            x + 2

    if y == 4 then
        (y, 1)
    else
        (y, 2)
```

Your suggested error would imply that the branches of the `y` expression should both be `(Num *, Num *)` even though the branches of `y` are actually `Num *`. 

I'm not sure of a better way to express this, so I'll leave it for now.

Reply:
I'm not sure I follow, those are not the branches of the function, they are the branches of a def inside the function.

Reply:
I agree. I think since "branches" refers usually to if/when branches, referring to the function's "branches" is a little confusing I expect, even if I personally can figure out what you mean. It seems like the most precise phrasing would be "every place your function can return from should return XYZ". Would that be worse or better in your eyes?